packages/discord.js/src/structures/Message.js

```diff
@@ -915,21 +920,39 @@ class Message extends Base {
    *   .then(() => console.log(`Replied to message "${message.content}"`))
    *   .catch(console.error);
    */
-  async reply(options) {
-    if (!this.channel) throw new DiscordjsError(ErrorCodes.ChannelNotCached);
+  reply(options) {
     let data;
 
     if (options instanceof MessagePayload) {
       data = options;
     } else {
       data = MessagePayload.create(this, options, {
-        reply: {
-          messageReference: this,
+        messageReference: {
+          messageId: this.id,
+          channelId: this.channelId,
+          guildId: this.guildId,
+          type: MessageReferenceType.Default,
           failIfNotExists: options?.failIfNotExists ?? this.client.options.failIfNotExists,
         },
       });
     }
-    return this.channel.send(data);
+    return this.client.channels.createMessage(this.channelId, data);
+  }
+
+  /**
+   * Forwards this message.
+   * @param {TextChannelResolvable} channel The channel to forward this message to.
+   * @returns {Promise<Message>}
+   */
+  forward(channel) {
+    return this.client.channels.createMessage(channel, {
+      messageReference: {
+        messageId: this.id,
+        channelId: this.channelId,
+        guildId: this.guildId,
```

Comment:
This is not typed to support `null`, but we'd be passing `null` here if we're forwarding a DM message. Could cause unexpected things in the future.

https://github.com/discordjs/discord.js/blob/19d48f6d6cb8218b896ddf69be313ebe05a96996/packages/discord.js/typings/index.d.ts#L6363